### PR TITLE
ci: run tests with sanitizers on linux

### PR DESCRIPTION
**Pull request by mxinden**

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -74,6 +74,14 @@ jobs:
           RUST_LOG=trace cargo +${{ matrix.rust-toolchain }} llvm-cov nextest $BUILD_TYPE --no-fail-fast --lcov --output-path lcov.info
           cargo +${{ matrix.rust-toolchain }} bench --no-run
 
+      - name: Run tests with sanitizers
+        if: matrix.os == 'ubuntu-latest' && matrix.rust-toolchain == 'nightly'
```
Comment (larseggert) [inline .github/workflows/check.yml:78]: I guess sanitizers only work on Linux?

Comment (mxinden) [reply .github/workflows/check.yml:78]: I am not sure. Looking into it now. 

https://doc.rust-lang.org/stable/unstable-book/compiler-flags/sanitizer.html

**Comment by larseggert** [comment]
We might want to add this to neqo once it works here.

File: .github/workflows/check.yml (hunk 1 above)
Comment (mxinden) [reply .github/workflows/check.yml]: All sanitizers are available on Linux and some on MacOS. I don't know how to run any of them on Windows.

**Review by larseggert — APPROVED** [review]

**Comment by mxinden** [comment]
> We might want to add this to neqo once it works here.

I can follow-up once this pull request is merged here.

**Pull request merged**